### PR TITLE
Add orphan secrets metric

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/8187

Due to a bug in app-operator we were leaving orphan secrets as well as configmaps. This makes sure both configmaps and secrets are being cleaned up correctly.

### CODE REVIEW

File: service/collector/orphan_configmap.go [hunk 1]
```diff
@@ -57,8 +57,6 @@ func NewOrphanConfigMap(config OrphanConfigMapConfig) (*OrphanConfigMap, error)
 }
 
 func (oc *OrphanConfigMap) Collect(ch chan<- prometheus.Metric) error {
-	oc.logger.Log("level", "debug", "message", "collecting metrics for orphan configmaps")
-
```
Comment (rossf7) [inline service/collector/orphan_configmap.go]: Making the logs less noisy. I'll do a follow up PR for the other metrics.

**Comment by rossf7** [comment]
Tested on `ghost`.

File: service/collector/orphan_secret.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+func (oc *OrphanSecret) Collect(ch chan<- prometheus.Metric) error {
+	charts, err := oc.g8sClient.ApplicationV1alpha1().Charts("").List(metav1.ListOptions{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	desiredSecrets := make(map[[2]string]bool)
+
+	for _, chart := range charts.Items {
+		desiredSecrets[[2]string{key.SecretNamespace(chart), key.SecretName(chart)}] = true
```
Comment (pipo02mix) [inline service/collector/orphan_secret.go]: cool I did not know it was possible in go :)

**Comment by rossf7** [comment]
Thanks @anvddriesch 👍 and I tested it on ghost earlier.

Aliyun job failed due to connectivity. Going with this.

File: service/collector/orphan_secret.go (hunk 2 above)
Comment (rossf7) [reply service/collector/orphan_secret.go]: This was magic from Mr Pawel.

**Pull request merged**